packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:ui';
+
+import '../../../components.dart';
+import '../../palette.dart';
+
+const _kMain = 'main';
+
+/// Adds a collection of paints to a component
+///
+/// Component will always have a main Paint that can be accessed
+/// by the [paint] getter/setter and other paints can be manipulated/accessed
+/// using [getPaint], [setPaint] and [deletePaint]
+mixin HasPaint on BaseComponent {
+  final Map<String, Paint> _paints = {
+    'main': BasicPalette.white.paint(),
```

Comment:
constant?

Reply:
Can't have constant in a method call like this

Reply:
I think you can, just tested on dartpad. it's not a method call, just a map literal initializer

![image](https://user-images.githubusercontent.com/882703/126991462-bf8b4678-1672-4c70-8448-d33a0ccc79cb.png)

but you can use constants on method calls too, in fact, anywhere that you can use a variable, you can use a constant, afaik

Reply:
`BasicPalette.white.paint()` is a method call and used to initialize the value of the map, so we can't use const here

Reply:
I don't think being a method call makes any difference, in fact:

```dart
const constant = 'key';

String methodCall() {
  return 'dynamic content';
}

class A {
  final Map<String, String> foo = {
    constant: methodCall(),
  };
}

void main() {
  print(A().foo[constant]);
}
```

works just fine. and I don't think there is any difference between `methodCall()` and `BasicPalette.white.paint()`...

Reply:
lol, now I get it, you want the key to be the constant, I thought you wanted to make the whole map const. Either way, I think I will change this as a whole because of our discussion on the type safe thing.

Reply:
yes, agree, that will totally bypass the magic string thing (if it works), but just wanted to comment this too in case that doesn't work too well